adobe/spectrum-web-components
# Update to Storybook@7.0 via @web/storybook-builder

## Description
- Adopt Storybook 7.0
  - unlocks federation
  - surfaces all Add-ons for consumption
  - speeds up builds/serving
- Move from `.storybook` to `storybook` to ensure publication to Netlify
- Include CSS storybook at dev time via federation
  - makes comparisons between the project easier than ever
  - exclude it at publication time

## How has this been tested?
    1. Go [here](https://storybook--spectrum-web-components.netlify.app/storybook/)
    2. Experience the amazingness that is our storybook
    1. Switch to this branch
    2. `rm -rf node_modules`, the graph resolves funky otherwise
    3. `yarn`
    4. Run storybook locally, `yarn storybook`
    5. See that the CSS storybook is federated

## Screenshots (if appropriate)
<img width="221" alt="image" src="https://github.com/adobe/spectrum-web-components/assets/1156657/c2f10d78-63ba-47e5-b4d3-4e6c05e76550">

## Types of changes

## Checklist
